### PR TITLE
Add label selector flag to get cmd

### PR DESCRIPTION
**Pull request by somtochiama**
This pull request adds a label selector flag to `get` command to enable users filter objects by label
e.g

```
flux get sources git -l sharding.fluxcd.io/key=shard1
```

### CODE REVIEW

File: cmd/flux/get.go [hunk 1]
```diff
@@ -150,6 +153,21 @@ func (get getCommand) run(cmd *cobra.Command, args []string) error {
 		listOpts = append(listOpts, client.MatchingFields{"metadata.name": args[0]})
 	}
 
+	if getArgs.labelSelector != "" {
+		label, err := metav1.ParseToLabelSelector(getArgs.labelSelector)
+		if err != nil {
+			return err
```
Comment (makkes) [inline cmd/flux/get.go:159]: I personally prefer wrapping errors and provide some context before returning them. But that may be personal preference.

**Review by makkes — APPROVED** [review]
Looks great.

File: cmd/flux/get.go (hunk 1 above)
Comment (stefanprodan) [reply cmd/flux/get.go:159]: Yes let's return here something like `invalid label expression, error: %w`

File: cmd/flux/get.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ func (get getCommand) run(cmd *cobra.Command, args []string) error {
 	if getArgs.labelSelector != "" {
 		label, err := metav1.ParseToLabelSelector(getArgs.labelSelector)
 		if err != nil {
-			return err
+			return fmt.Errorf("unable to parse label selector: error: %w", err)
```
Comment (makkes) [inline cmd/flux/get.go:159]:
```suggestion
			return fmt.Errorf("unable to parse label selector: %w", err)
```

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @somtochiama this will help with sharding 🥇

**Pull request merged**